include/kernels/contraction.cuh

```diff
@@ -8,14 +8,490 @@
 
 namespace quda
 {
+  static constexpr int max_contract_results = 16; // sized for nSpin**2 = 16
+ 
+  using spinor_array = array<array<double, 2>, max_contract_results>;
+  using staggered_spinor_array = array<double, 2>; 
+
+  template <typename real, int nSpin = 4> class DRGammaMatrix {
```

Comment:
Should this class definition be in its own header?

Reply:
@maddyscientist the class is used only in this file, so is there any other reason to isolate it?

Reply:
Addressed in 4d76007